src/node_task_queue.cc

```diff
@@ -76,12 +114,39 @@ void PromiseRejectCallback(PromiseRejectMessage message) {
 
   Local<Value> args[] = { type, promise, value };
 
-  // V8 does not expect this callback to have a scheduled exceptions once it
-  // returns, so we print them out in a best effort to do something about it
-  // without failing silently and without crashing the process.
+  double async_id = AsyncWrap::kInvalidAsyncId;
+  double trigger_async_id = AsyncWrap::kInvalidAsyncId;
+  GetAssignedPromiseAsyncId(env, promise, env->async_id_symbol())
+          .To(&async_id);
+  GetAssignedPromiseAsyncId(env, promise, env->trigger_async_id_symbol())
+          .To(&trigger_async_id);
+  
+  GetAssignedPromiseWrapAsyncId(env, promise, env->async_id_symbol())
+            .To(&async_id);
+  GetAssignedPromiseWrapAsyncId(env, promise, env->trigger_async_id_symbol())
+          .To(&trigger_async_id);
```

Comment:
These calls can lead to JS exceptions, so I think you want to move all of this into the `TryCatchScope` here and return if any of these `.To()` calls return `false`